### DISCUSSION TITLE
Using API tokens (private runtimeConfig) in client-side

### DISCUSSION THREAD

**Discussion by shahzodjon-jumakulov** [nuxt/nuxt#20544] [Questions] [answered]
nuxt.config.ts
```
runtimeConfig: {
  public: {
    BASE_URL: process.env.BASE_URL,
  },
  API_TOKEN: process.env.API_TOKEN
},
```

I am using useFetch() like this:
```
useFetch("url", {
  baseURL: config.public.BASE_URL,
  headers: {
    "X-...-API-KEY": config.API_TOKEN,
  }
});
```
On the first load of the page, everything works fine. But after navigating to other pages, useFetch() function can't use API token (useRuntimeConfig().API_TOKEN). How can I get access to the private runtimeConfigs on client-side? Or is there any other way to solve this?

**Comment by maou-shonen** [comment, marked as answer]
You cannot use private runtimeConfig on client,
The most common example on nuxt3 is that the client fetches to the server, and then the server sends an external request,

Here is a simple example.

```vue
// app.vue
<script setup>
useFetch("url", {
  baseURL: '/api/example',
});
</script>
```

```ts
// server/api/example
export default defineEventHandler(async (event) => {
  const config = useRuntimeConfig()
  return await $fetch(event.path || '/', {
    baseURL: config.api.baseUrl,
    method: getMethod(event),
    query: getQuery(event),
    headers: {
      'X-Token': config.api.token,
    }
  })
})
```

Of course, you should choose the appropriate solution according to your needs

**Reply by shahzodjon-jumakulov** [reply]
Thanks for your quick response)

**Reply by simon-ccoms** [reply]
Hi @maou-shonen - this looks like the solution for me too but it doesn't seem to work. It looks like I need a new file for every API request whereas I wanted to have one file on the server to catch all the requests and send them to the API. When you call this file "server/api/example" , was it meant to be called index.ts in that directory?

**Reply by maou-shonen** [reply]
`~/server/api/[...slug].ts`

You can also use h3 `sendRequestProxy` | `sendProxy` instead of fetch

**Reply by simon-ccoms** [reply]
Strange - when I do that, I get:-

[9:33:35 AM] [nuxt] [request error] [unhandled] [404]  (404 Not Found (<API base URL>/api/example/url))

So it's using the baseURL from the useFetch and appending the url from the request. I was assuming it would only use the URL I requested ("url") on the API?

**Reply by shahzodjon-jumakulov** [reply]
Hi there @simon-ccoms ,

I suggest you to take a look at [this](https://stackoverflow.com/a/76134953/21239162) too. Hope it helps

**Reply by simon-ccoms** [reply]
Ah, so instead of event.path, if you use event.context.params!.path then it works! Thanks!
